_posts/2020-10-22-modele-wdrozenia-w-chmurze.md

```diff
@@ -0,0 +1,199 @@
+- abstrakcję infrastruktury,
+- minimalny koszt wdrożenia i utrzymania aplikacji,
+- przeniesienie ciężaru na rozwój funkcjonalny aplikacji,
+- dostępność gotowych do integracji z systemem usług w modelu SaaS
+
+# Vendor lock-in
+
+Planując migrację do chmury musisz zmierzyć się z tematem uzależnienia kodu od konkretnego dostawcy.
+
+Czy _vendor lock-in_ jest zły? Zagadnienie warto traktować jako ryzyko, a nie zagrożenie jako takie. Dla ryzyk możliwe jest zaplanowanie ścieżek przeciwdziałania i przygotowanie planu wyjścia z chmury (czy to do innego dostawcy, czy zupełnie w kierunku chmury prywatnej lub rozwiązania _on-premises_).
+
+Dla podejść IaaS i CaaS możemy przyjąć, że ryzyko wyjścia z chmury niesie za sobą minimalne koszty - w szczególności, gdy system nie korzysta z natywnych usług oferowanych przez chmurę obliczeniową.
+
+W przypadku PaaS i FaaS prawdopodobnie zaprojektowałeś system pod kątem oferty konkretnej chmury. Najwięksi dostawcy zapewniają ujednoliconą ofertę produktów. O ile różnice pomiędzy oferowanymi rozwiązaniami w wielu przypadkach uniemożliwiają migrację systemu bez żadnych zmian, to jednak znajdziemy pomiędzy nimi odpowiadające sobie rozwiązania pozwalające zrealizować system w podobnej architekturze.
+
+Warto zauważyć, że problem _vendor lock-in_, jako poważne ryzyko migracji do chmury, często jest adresowany wprost przez dostawców. Większość produktów Google Cloud Platform jest zbudowana w oparciu o otwarte standardy oraz narzędzia open-source. Przykładowo Google Kubernetes Engine do działania wykorzystuje platformę Kubernetes, Cloud Bigtable jest zgodny z HBase, itd.
+
+Szacując koszty wyjścia z chmury powinieneś rozważyć również utracony zyski wynikające z obaw przed _vendor lock-in_. Obawa przed uzależnieniem może doprowadzić do zwiększenia złożoności systemu, zwiększenia kosztu produkcji i utrzymania systemu, utraty funkcjonalności dających przewagę, itp. Czy ew. koszt wyjścia z chmury będzie większy niż oszczędności wynikające z wykorzystania chmury? W przypadku dużego prawdopodobieństwa lub wysokiego kosztu wyjścia z chmury warto rozważyć ograniczenie migracji do modelu CaaS.
+
+# On-prem, IaaS, CaaS, PaaS, FaaS
+
+Mając już w głowie argumentacje i przykłady wszystkich podejść, możemy nakreślić podsumowanie odpowiedzialności w różnych modelach.
+
+![Porównanie modeli wdrożeń](/assets/img/posts/2020-10-22-modele-wdrozenia-w-chmurze/compare-xaas.png)
+
+Im bardziej zaadaptujemy rozwiązania chmurowe, tym więcej odpowiedzialności przerzucimy na stronę dostawcy.
+
+# Dojrzałość chmurowa
+
+Znając system, możemy pokusić się o ocenę jego dojrzałości chmurowej. Z jednej strony mamy klasyczne rozwiązania _on-premises_, a z drugiej rozwiązania w pełni serverless. Podsumowując wady i zalety opisywanych do tej pory podejść możemy pokusić się o prosty diagram szans i ryzyk zależny od stopnia migracji do chmury:
```

Comment:
Wydaje mi się, że przecinek po "Znając system" jest zbędny. W dwóch sąsiadujących zdaniach jest to samo wyrażenie "możemy pokusić się", co trochę razi